docs/en/02_Developer_Guides/14_Files/03_File_Security.md

```diff
@@ -2,13 +2,85 @@ summary: Manage access permission to assets
+ * "Protected file": A file with access restrictions.
+   Note that draft files are always protected, and even published files
+   can be protected if they have access restrictions.
+
+## Permission Model
+
+Like all other objects in SilverStripe, permissions are generally controlled via `can*()` methods,
+for example `canView()` (see [permissions](/developer_guides/security/permissions)).
+
+The permission model defines the following actions:
+
+ * View: Access file metadata in the database
+ * Edit: Edit file metadata as well as replacing the file content
+ * Create: Create file metadata and upload file content
+ * Delete: Delete file metadata and the file content
+ * Download: Access the file content, but not the file metadata.
+   Usually treated the same as "View".
+
+There's a few rules guiding their access, in descending order of priority:
+
+ * Published and unprotected files can be downloaded by anyone knowing the URL.
+   They bypass any SilverStripe permission checks (served directly by the webserver). 
+ * Access can be restricted by custom `can*()` method implementations on `File`
+   (through [extensions](/developer_guides/extending/extensions)).
+   This logic can overrule any further restrictions below.  
+ * Users with "Full administrative rights" (`ADMIN` permission code)
+   have view and edit access by default, regardless of further restrictions below.
+ * Users with "Edit any file" permissions (`FILE_EDIT_ALL` permission code)
+   have edit access by default, regardless of further restrictions below.
+ * View or edit access can be restricted per file or folder through
+   an inherited permissions model similar to page content (through [api:SilverStripe\Security\InheritedPermissionsExtension]).
+   There are four types: "Inherit from parent" (default), "Anyone", "Logged-in users", or "Only these groups".
+ * Protected files (incl. draft files) allow view/edit access when `File::$non_live_permissions` is satisfied.
+   By default, that's configured for anyone with access to any CMS section, or
+   the ability to "view draft content".
+ * Protected files need an "access grant" for the current session
+   in order to download the file (see [User access control](#user-access-control)).
+   While you can technically allow viewing or editing a file without granting
+   access to download it, those aspects are usually bundled together by the file viewing logic. 
+
+Access to create or delete files generally aligns with the edit access described above.
+
+Note that even if the permissions above allow access,
+you need to have access to a mechanism to view or edit file information. 
+Most commonly this is through the "Access to Files section" permission.
+Custom implementations (e.g. APIs or custom file viewers) can have
+further restrictions in your project. 
+
+<div class="warning" markdown="1">
+When implenting your own `canView()` logic through [extensions](/developer_guides/extending/extensions),
```

Comment:
FYI I don't think these HTML sections get parsed as markdown right now, so this link would get rendered literally

Reply:
We have lots of other sections like that in the docs: https://github.com/silverstripe/silverstripe-framework/search?q=%3Cdiv+class%3D%22warning%22+markdown%3D%221%22%3E&unscoped_q=%3Cdiv+class%3D%22warning%22+markdown%3D%221%22%3E. They render just fine: https://docs.silverstripe.org/en/4/contributing/documentation

Reply:
Oh wait, you mean the *link* within that ... works fine as well: https://docs.silverstripe.org/en/4/developer_guides/model/versioning (see "The extension is automatically applied...")

Reply:
OK, sweet!